pages/[endpoint]/block.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { useQuery } from '@apollo/client';
+
+import Link from 'components/Link';
+import { TransactionList } from 'components/List';
+import Timestamp from 'components/Timestamp';
+import { getEndpointFromQuery, GRAPHQL_ENDPOINTS } from 'lib/graphQLEndPoint';
+import { listTxColumns } from 'lib/listColumns';
+import {
+  CommonPageProps,
+  getCommonStaticPaths as getStaticPaths,
+  getCommonStaticProps as getStaticProps,
+} from 'lib/staticGeneration';
+import useIdFromQuery from 'lib/useIdFromQuery';
+import useSearchParams from 'lib/useSearchParams';
+
+import {
+  Block,
+  BlockByHashDocument,
+  BlockByHashQuery,
+  Transaction,
+} from 'src/gql/graphql';
+
+export default function BlockPage({ staticEndpoint }: CommonPageProps) {
+  const [pageContent, setPageContent] = useState<JSX.Element>(<></>);
+
+  const [endpoint, setEndpoint] = useState(staticEndpoint);
+  const { isReady, asPath } = useRouter();
+  const [query] = useSearchParams(asPath);
+  const hash = useIdFromQuery(query);
```

Comment:
Currently uri format is `/block/?[blockhash]`
I think it would be better to change like this `/block/[blockhash]`

You just need to move this file /[endpoint]/block/[blockhash].tsx
and you can read block hash from `useRouter()`

Reply:
This is intentionally done in this way to allow for static site version of the app, which is required to embed into ASP.Net applications.